### PR TITLE
Stats API: allow escaping `|` literal character with `\|` when filtering

### PR DESCRIPTION
**Pull request by RobertJoonas**
### Changes

- Allow escaping the `|` character with `\|` in Stats API filter values, to prevent the filter type being parsed to `:member`

### Tests
- [x] Automated tests have been added

### Changelog
- [x] Entry has been added to changelog

### Documentation
- [x] Documentation update PR is pending: https://github.com/plausible/docs/pull/279

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

**Review by aerosol — APPROVED** [review]
Looks fine, it's great that we have parser-level tests now so we'll keep track of regressions moving forward :heart:

File: lib/plausible/stats/filter_parser.ex [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  ### Examples:
+
+      iex> FilterParser.parse_filters(%{"filters" => "{\\"page\\":\\"/blog/**\\"}"})
+      %{"page" => "/blog/**"}
+
+      iex> FilterParser.parse_filters(%{"filters" => "visit:browser!=Chrome"})
+      %{"visit:browser" => {:is_not, "Chrome"}}
+  """
+  def parse_filters(%{"filters" => filters}) when is_binary(filters) do
```
Comment (ukutaht) [inline lib/plausible/stats/filter_parser.ex:25]: Only thing that bothers me a bit is this module taking a Map as input. Feels like it's concerned with the internals of the `Query` struct and doing something that's unrelated to filter parsing. I think the separation would be cleaner if this module just took the value of the `filters` key:

```elixir
# in query.ex
FilterParser.parse_filters(query["filters"])
```

```elixir
# in filter_parser.ex
def parse_filters(filters) when is_binary(filters) do
   ...
end
```

**Review by ukutaht — COMMENTED** [review]
Looks good! One comment inline

File: lib/plausible/stats/filter_parser.ex (hunk 1 above)
Comment (RobertJoonas) [reply lib/plausible/stats/filter_parser.ex]: Good point! Did it now: https://github.com/plausible/analytics/pull/2266/commits/6ae6a8f3d9e6c0235ca69507e662ca56813e3e1f

**Review by ukutaht — APPROVED** [review]

**Pull request merged**